stream-chat-android-client/src/main/java/io/getstream/chat/android/client/uploader/FileUploader.kt

```diff
@@ -19,7 +19,7 @@ public interface FileUploader {
         connectionId: String,
         file: File,
         callback: ProgressCallback
-    )
+    ): String?
```

Comment:
The old implementation here was a fire and forget call for sending a file, likely not useful for anyone (and it gave no opportunity to handle errors).

Making this synchronous instead of async is a breaking API change, but we don't expect anyone to be calling the `FileUploader` APIs directly. We also don't expect any existing implementations other than our own, but those will break at compile time, and will have to be updated to be blocking.

(Still, we should probably add a line in the changelog about this.)

Reply:
`FileUploader` could be implemented by our customers to upload files to their own server

Reply:
Those implementations will indeed break and have to be updated (should be fairly simple), but at least they won't get a silent behaviour change.

We're also betting on not many customers having done this yet, as it's pretty new API.

We pretty much can't fix this without a breaking change for existing implementations, as introducing a new method in the interface with a different name would also be a breaking change, and we can't adapt a fire-and-forget async API into a sync one, so a default implementation can't be provided either.

Reply:
Yep, I agree, I wanted to let you all know that it is a public interface because it can be (and it is)  implemented by our customers, and what is the reason they would like to implement it.
The breaking change is easy to fix the when they will need to update the integration with our SDK